### ISSUE TITLE
Unexpected behavior: calc_ps() after changing EVP parameters doesn't provide new evalues

### ISSUE THREAD

**Issue by afraser3** [DedalusProject/eigentools#9]
I'm dealing with a use case where I want to calculate the pseudospectrum over a range of physical parameters. Strangely, `calc_ps()` gives me new pseudospectra at each parameter point as expected, but not the actual spectrum.

Steps to reproduce:

1. Create an EVP
2. Create an EP from that EVP
3. Do `EP.calc_ps()` ([note that this also calls `EP.solve`](https://github.com/DedalusProject/eigentools/blob/d875acdb4ce86027f60bc6f4516ae917d6ac0062/eigentools/eigenproblem.py#L393)) and extract/save `EP.evalues` and `EP.pseudospectrum`
4. Change a physical parameter for the EVP by doing `problem.namespace['parameter_name'].value = new_value`
5. Do `EP.calc_ps()` again and extract/save `EP.values` and `EP.pseudospectrum`
6. Note that the new `EP.pseudospectrum` has been appropriately updated, while the new `EP.values` is now an empty list.

I'm able to circumvent this by using `EP._set_parameters()`, but it sounds like that's not the intended way to go about this.

[Working example is here](https://github.com/DedalusProject/eigentools/files/6826450/MHD_OrrSomm_pseudospec_issue_illustration.py.zip) (I had to .zip it so that github would accept it). I changed `problem.parameters` in addition to `problem.namespace` because I saw that's ultimately what `_set_parameters()` does ([as seen here](https://github.com/DedalusProject/eigentools/blob/d875acdb4ce86027f60bc6f4516ae917d6ac0062/eigentools/tools.py#L6)) but still no luck. This leaves me thinking that what's probably happening is the new evalues are all being rejected because I didn't manually redo the high-res spectrum.

**Comment by jsoishi** [comment]
Hi @afraser3,

This is a design bug on our part. The reason your code doesn't work is because the new version of eigentools expects parameters to be set by passing a dict to `.solve()`,  but we didn't provide such an option to `.calc_ps()`. I think it is fixed in commit bb59ce8. Could you please give it a try? The only modification to your script is to delete 
```
problem.namespace['MA2'].value = 1.75 ** 2.0
problem.parameters['MA2'] = 1.75 ** 2.0`
```
and replace the second call to `.calc_ps()` with this:

```
EP.calc_ps(k, (real_points, imag_points), inner_product=energy_norm, parameters={'MA2':1.75**2})
```
Also, I noticed your example fails to converge for many cases on my machine. Not sure if this is true on yours.

**Comment by afraser3** [comment]
Works like a charm! Thanks for looking into it so quickly.
